WooCommerce/Classes/ViewRelated/Products/Add Product/ManualProductTypeOptions.swift

```diff
@@ -6,37 +6,70 @@ struct ManualProductTypeOptions: View {
     private let productTypes: [BottomSheetProductType]
     private let onOptionSelected: (BottomSheetProductType) -> Void
 
+    /// Grouped product types based on their product category
+    private let groupedProductTypes: [(BottomSheetProductType.ProductCreationCategory, [BottomSheetProductType])]
+
     init(productTypes: [BottomSheetProductType],
          onOptionSelected: @escaping (BottomSheetProductType) -> Void) {
         self.productTypes = productTypes
         self.onOptionSelected = onOptionSelected
+
+        self.groupedProductTypes = Constants.productCategoriesOrder.compactMap { category in
+            let currentCategoryProductTypes = productTypes.filter { $0.productCreationCategory == category }
+            return currentCategoryProductTypes.isEmpty ? nil : (category, currentCategoryProductTypes)
+        }
     }
 
     var body: some View {
-        ForEach(productTypes) { productType in
-            HStack(alignment: .top, spacing: Constants.margin) {
-                Image(uiImage: productType.actionSheetImage.withRenderingMode(.alwaysTemplate))
-                    .font(.title3)
-                    .foregroundStyle(.secondary)
-
-                VStack(alignment: .leading, spacing: Constants.verticalSpacing) {
-                    Text(productType.actionSheetTitle)
-                        .bodyStyle()
-                    Text(productType.actionSheetDescription)
-                        .subheadlineStyle()
+        ForEach(groupedProductTypes, id: \.0) { (category, productTypes) in
+            VStack {
+                Text(category.label)
+                    .subheadlineStyle()
+                    .textCase(.uppercase)
+                    .frame(maxWidth: .infinity, alignment: .leading)
+                    .padding(.bottom, Constants.categoryVerticalSpacing)
+                    .padding(.horizontal, Constants.horizontalSpacing)
+
+                ForEach(productTypes, id: \.self) { productType in
+                    HStack(alignment: .top, spacing: Constants.margin) {
+                        Image(uiImage: productType.actionSheetImage.withRenderingMode(.alwaysTemplate))
+                            .font(.title3)
```

Comment:
The modifier `font` will work only with system images. With custom assets, you might need to set the size explicitly with the`frame` modifier. To make it scalable, you'd need to get the scale from `ScaleMetric` property wrapper.

<img src="https://github.com/woocommerce/woocommerce-ios/assets/5533851/c6ecb122-0f55-4df4-93d8-2178ad9a88dc" width=320 />

As you see above, the custom icons are not scaled in large font sizes.

Reply:
Updated in https://github.com/woocommerce/woocommerce-ios/pull/12273/commits/34442d134d84058e62685dcfb68a6ecdfc1bae34